testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiMinioReads.java

```diff
@@ -0,0 +1,240 @@
+ * limitations under the License.
+ */
+package io.trino.tests.product.hudi;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.HUDI;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onHudi;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestHudiMinioReads
+        extends ProductTest
+{
+    protected String bucketName;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        bucketName = requireNonNull(System.getenv("S3_BUCKET"), "Environment variable not set: S3_BUCKET");
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteTableSelect()
+    {
+        String tableName = "test_hudi_demo_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createNonPartitionedHudiTable(tableName, tableDirectory, bucketName);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "Trino"),
+                row(2, "rocks"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT uuid, col FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT uuid, col FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWritePartitionedTableSelect()
+    {
+        String tableName = "test_hudi_demo_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        onHudi().executeQuery(format("CREATE TABLE default.%s (\n" +
+                        "  id bigint,\n" +
+                        "  name string,\n" +
+                        "  ts bigint,\n" +
+                        "  dt string,\n" +
+                        "  hh string\n" +
+                        ") USING hudi\n" +
+                        "tblproperties (\n" +
+                        "  type = 'cow',\n" +
+                        "  primaryKey = 'id',\n" +
+                        "  preCombineField = 'ts'\n" +
+                        " )\n" +
+                        "partitioned by (dt, hh) LOCATION 's3://%s/%s'",
```

Comment:
Improve the format:
```suggestion
                        "USING hudi " +
                        "TBLPROPERTIES (" +
                        "  type = 'cow'," +
                        "  primaryKey = 'id'," +
                        "  preCombineField = 'ts'" +
                        " )" +
                        "PARTITIONED BY (dt, hh) " +
                        "LOCATION 's3://%s/%s'",
```

testMergeOnReadTableSelect needs similar change.